tests/integration/records/delete-record-test.js

```diff
@@ -136,6 +136,77 @@ test('deleting a record that is part of a hasMany removes it from the hasMany re
   assert.equal(group.get('people.length'), 1, 'expected 1 related records after delete');
 });
 
+test('deleting a record that is part of a hasMany removes it from the hasMany recordArray when adapter deleteRecord function returns a payload', function(assert) {
+  let group;
+  let person;
+  const Group = DS.Model.extend({
+    people: DS.hasMany('person', { inverse: null, async: false })
+  });
+  Group.toString = () => { return 'Group'; }
+
+  env.adapter.deleteRecord = function() {
+    return EmberPromise.resolve({
+      data: {
```

Comment:
it's important to note that no matter what this value is, as long as it is not `undefined`, the test seems to still fail.